linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_name_format/ansible/shared.yml

```diff
@@ -10,13 +10,32 @@
   {{%- set auditd_conf_path=audisp_conf_path + "/auditd.conf" %}}
 {{%- endif %}}
 
-
 {{{ ansible_instantiate_variables("var_auditd_name_format") }}}
 
-{{{ ansible_set_config_file(file=auditd_conf_path,
-                  parameter="name_format",
-                  value="{{ var_auditd_name_format }}",
-                  create=true,
-                  separator=" = ",
-                  separator_regex="\s*=\s*",
-                  prefix_regex="(?i)^\s*") }}}
+- name: Set type of computer node name logging in audit logs
+  block:
+  - name: Check for duplicate values
+    lineinfile:
+      path: "{{{ auditd_conf_path }}}"
+      create: true
+      regexp: (?i)^\s*name_format\s*=\s*
+      state: absent
+    check_mode: true
+    changed_when: false
+    register: dupes
+
+  - name: Deduplicate values from {{{ auditd_conf_path }}}
+    lineinfile:
+      path: "{{{ auditd_conf_path }}}"
+      create: true
+      regexp: (?i)^\s*name_format\s*=\s*
+      state: absent
+    when: dupes.found is defined and dupes.found > 1
+
+  - name: Insert correct line to {{{ auditd_conf_path }}}
+    lineinfile:
+      path: "{{{ auditd_conf_path }}}"
+      create: true
+      regexp: (?i)^\s*name_format\s*=\s*
+      line: name_format = {{ var_auditd_name_format.split('|').[0] }}
```

Comment:
The macro is different only in this line, if I am not wrong. Maybe you can update the `ansible_set_config_file` macro before calling the `ansible_only_lineinfile` ? Have you considered this option? If would be great because more rules using this macros would also benefit on this, making them easier to be updated in the future. It would also reduce duplication.

Reply:
Or even more simpler, you could use a similar approach used in Bash, with `set_fact` for example.